.github/workflows/misc-linters.yml

```diff
@@ -33,3 +33,32 @@ jobs:
           pip install codespell
           # exclude files which may be synchronized from other places
           git ls-files | grep -v "^thirdparty" | grep -v "/thirdparty/" | grep -v "/dist/" | xargs -t codespell --ignore-words .github/workflows/ignore_words 2>/dev/null
+  lineguard:
+    name: lineguard
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - name: Harden the runner (Audit all outbound calls)
+        uses: step-security/harden-runner@6c439dc8bdf85cadbbce9ed30d1c7b959517bc49 # v2.12.2
+        with:
+          egress-policy: audit
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          fetch-depth: 0
+      - name: Ensure git safe directory
+        run: |
+          git config --global --add safe.directory $(pwd)
+      - name: Install lineguard
+        run: |
+          curl -L https://github.com/hydai/lineguard/releases/download/v0.1.3/lineguard-v0.1.3-x86_64-unknown-linux-gnu.tar.gz | tar xz
+          sudo mv lineguard /usr/local/bin/
+      - name: Run lineguard
+        run: |
+          BASE_REF=${{ github.base_ref }}
+          if git rev-parse --verify origin/$BASE_REF >/dev/null 2>&1; then
+            lineguard --config .lineguardrc --fix --from origin/$BASE_REF --to HEAD || { echo "Lineguard found issues"; exit 1; }
+          else
+            lineguard --config .lineguardrc --fix --from HEAD~1 --to HEAD || { echo "Lineguard found issues"; exit 1; }
```

Comment:
You should use `git rev-parse` to obtain the commit hash and use it for the lineguard.